src/library_wasmfs.js

```diff
@@ -175,7 +175,65 @@ FS.createPreloadedFile = FS_createPreloadedFile;
     },
     // TODO: readlink
     // TODO: stat
+    stat: (path) => {
+      return withStackSave(() => {
+        var pathBuffer = stringToUTF8OnStack(path);
+
+        var statBuf = _malloc({{{ C_STRUCTS.stat.__size__ }}});
+        FS.handleError(__wasmfs_stat(pathBuffer, statBuf));
+        _free(statBuf);
+
+        // i53/u53 are enough for times and ino in practice.
+        var stats = {
+          dev: {{{ makeGetValue('statBuf', C_STRUCTS.stat.st_dev, "u32") }}},
+          mode: {{{ makeGetValue('statBuf', C_STRUCTS.stat.st_mode, "u32") }}},
+          nlink: {{{ makeGetValue('statBuf', C_STRUCTS.stat.st_nlink, "u32") }}},
+          uid: {{{ makeGetValue('statBuf', C_STRUCTS.stat.st_uid, "u32") }}},
+          gid: {{{ makeGetValue('statBuf', C_STRUCTS.stat.st_gid, "u32") }}},
+          rdev: {{{ makeGetValue('statBuf', C_STRUCTS.stat.st_rdev, "u32") }}},
+          size: {{{ makeGetValue('statBuf', C_STRUCTS.stat.st_size, "i53") }}},
+          blksize: {{{ makeGetValue('statBuf', C_STRUCTS.stat.st_blksize, "u32") }}},
+          blocks: {{{ makeGetValue('statBuf', C_STRUCTS.stat.st_blocks, "u32") }}},
+          atime: {{{ makeGetValue('statBuf', C_STRUCTS.stat.st_atim.tv_sec, "i53") }}},
+          mtime: {{{ makeGetValue('statBuf', C_STRUCTS.stat.st_mtim.tv_sec, "i53") }}},
+          ctime: {{{ makeGetValue('statBuf', C_STRUCTS.stat.st_ctim.tv_sec, "i53") }}},
+          // TODO: st_ino should be unsigned, but there is an error using u53
+          ino: {{{ makeGetValue('statBuf', C_STRUCTS.stat.st_ino, "i53") }}}
```

Comment:
Is it a missing function dependency? Adding that dependency around line 24 should fix that.